Differences in mypy output between pre-commit and VSCode

(@fzimmermann89 I moved this discussion to a new issue)

When I run 
`pre-commit run --all-files `
in the mrpro folder then the mypy checks all pass.

When I run 
`mypy src/mrpro `
(which is I think what VS Code is running) then I get:
_src/mrpro/data/_SpatialDimension.py:88: error: Incompatible return value type (got "SpatialDimension[int]", expected "SpatialDimension[Tensor]")  [return-value]
src/mrpro/data/_SpatialDimension.py:88: error: Incompatible return value type (got "SpatialDimension[float]", expected "SpatialDimension[Tensor]")  [return-value]
src/mrpro/data/_SpatialDimension.py:88: error: Argument 1 to "SpatialDimension" has incompatible type "Tensor"; expected "int"  [arg-type]
src/mrpro/data/_SpatialDimension.py:88: error: Argument 1 to "SpatialDimension" has incompatible type "Tensor"; expected "float"  [arg-type]
src/mrpro/data/_SpatialDimension.py:88: error: Argument 2 to "SpatialDimension" has incompatible type "Tensor"; expected "int"  [arg-type]
src/mrpro/data/_SpatialDimension.py:88: error: Argument 2 to "SpatialDimension" has incompatible type "Tensor"; expected "float"  [arg-type]
src/mrpro/data/_SpatialDimension.py:88: error: Argument 3 to "SpatialDimension" has incompatible type "Tensor"; expected "int"  [arg-type]
src/mrpro/data/_SpatialDimension.py:88: error: Argument 3 to "SpatialDimension" has incompatible type "Tensor"; expected "float"  [arg-type]
Found 8 errors in 1 file (checked 12 source files)_

Does anybody know what the pre-commit is doing differently with mypy?

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -28,35 +28,40 @@ repos:
       - id: autopep8
 
   - repo: https://github.com/PyCQA/flake8
-    rev: 6.0.0
+    rev: 6.1.0
     hooks:
       - id: flake8
         args: [--exclude=__init__.py]
         exclude: ^docs/
 
   - repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v0.950
+    rev: v1.5.1
     hooks:
       - id: mypy
+        additional_dependencies:
+          - numpy
+          - torch
+          - --index-url=https://download.pytorch.org/whl/cpu
+          - --extra-index-url=https://pypi.python.org/simple
         exclude: "docs/.*|tests/.*"
 
   - repo: https://github.com/myint/docformatter
-    rev: v1.4
+    rev: v1.7.5
     hooks:
       - id: docformatter
         args: ["--in-place"]
 
   - repo: https://github.com/PyCQA/pydocstyle
-    rev: 6.1.1
+    rev: 6.3.0
     hooks:
       - id: pydocstyle
         exclude: __init__.py
         files: ^src/
         additional_dependencies:
-          - toml
+          - tomli
 
   - repo: https://github.com/asottile/pyupgrade
-    rev: v3.9.0
+    rev: v3.10.1
     hooks:
     -   id: pyupgrade
         args: ["--py310-plus"]

